src/bpftrace.cpp

```diff
@@ -153,40 +161,24 @@ int BPFtrace::add_probe(ast::Probe &p)
           feature_->has_uprobe_multi() && has_wildcard(attach_point->func) &&
           !p.need_expansion && attach_funcs.size()) {
         if (!has_wildcard(attach_point->target)) {
-          Probe probe;
-          probe.attach_point = attach_point->func;
-          probe.path = attach_point->target;
-          probe.type = probetype(attach_point->provider);
-          probe.log_size = config_.get(ConfigKeyInt::log_size);
-          probe.orig_name = p.name();
-          probe.name = attach_point->name(attach_point->target,
-                                          attach_point->func);
-          probe.index = p.index();
+          auto probe = generate_probe(*attach_point, p);
           probe.funcs = attach_funcs;
-
           resources.probes.push_back(probe);
           continue;
         } else {
           // If we have a wildcard in the target path, we need to generate one
           // probe per expanded target.
           std::unordered_map<std::string, Probe> target_map;
           for (const auto &func : attach_funcs) {
-            std::string func_id = func;
-            std::string target = erase_prefix(func_id);
-            auto found = target_map.find(target);
+            ast::AttachPoint ap = attach_point->match_override(func);
+            ap.func = attach_point->func;
```

Comment:
What's this line for? It seems like `match_override()` replaces a wildcarded function name (e.g. "read*line") with an actual matched function name ("readline"). But then this line restores the wildcarded name ("read*line")

Reply:
Is `ap.func` actually used anywhere for uprobe_multi? Might be clearer to just zero it out if not

Reply:
It's used to generate the `Probe` name in `BPFtrace::generate_probe`. For now, that's only useful for debugging but the follow-up PRs will need the correct name to identify the correct BPF program.